enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,529 @@
+instead of host level settings, providing greater flexibility and being self
+contained within MicroShift.
+
+Users in need of blocking traffic shall use the `NetworkPolicy` resource. A
+section in the documentation will be added to describe how to do this.
+
+### Configure IPs where the router listens
+The following configuration is proposed:
+```yaml
+ingress:
+  expose: # Defaults to IPs in the host. Details below.
+    - <NIC name|IP address>
+    - ...
+```
+
+As described in [this section](#using-loadbalancer-service-type), the use of
+`LoadBalancer` makes ovnk configure iptables rules to expose the service
+outside of the cluster.
+Building on [LoadBalancer controller enhancement](loadbalancer-service-support.md),
+the service is exposed using the node IP and configured ports.
+
+All the IPs where the router is exposed need to be part of the `LoadBalancer`
+service status definition. In order to simplify configuration the router will
+allow specifying only IP addresses and interface names.
+
+MicroShift will default to listen to external IPs, the service IP and the
+apiserver IP, as this was the previous behavior and it needs to keep it for
+compatibility. Any advanced configuration, such as multiple interfaces, VLANs,
+etc. will need user's configuration. Any user's configuration will override all
+of the defaults.
+
+If the configuration includes duplicates in the form of same entries in the
+same list or referring to the same IP in different ways (hostnames or NICs),
+these will be ignored without warnings/errors.
+
+The NIC IP addresses will be automatically picked up by MicroShift's
+[service controller](loadbalancer-service-support.md). This is described in
+more details in [this section](#using-loadbalancer-service-type).
+
+### Allow specific IP addresses
+This may be seen as a special case of firewalling, but using IP addresses
+instead of ports. Using the same `NetworkPolicy` resources, IPs may be
+specified to allow traffic from them and reject it from any other source.
+
+Please note this is a manual procedure for the user. Documentation will be
+added on how to perform the required actions.
+
+More information in the design details section.
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for the router prior to
+   MicroShift's start.
+2. After MicroShift started, the system will ingest the configuration and setup
+   everything according to it.
+3. The router will be enabled/disabled, be exposed on the specified ports,
+   listen on the specified IPs, and allow connections from only specified the
+   IP addresses, all according to the cluster admin-provided configuration.
+   This is reflected in the status in the LoadBalancer-type service for the
+   router (if the router is enabled).
+
+### API Extensions
+As described in the proposal, there is an entire new section in the configuration:
+```yaml
+ingress:
```

Comment:
Which MicroShift API object is this?

Reply:
This is MicroShift's start configuration.

Reply:
Ah, ok.  Is it an object in openshift/api ? That's usually what `API Extensions` refers to in the proposal.

Reply:
Ah I see. In this case MicroShift is a bit different. This is the [spec](https://github.com/openshift/microshift/blob/main/cmd/generate-config/config/config-openapi-spec.json) for the configuration, but its not present anywhere else in OCP.